### PR TITLE
Don't use linked-list for contact pooling

### PR DESCRIPTION
**Pull request by metalgearsloth**
Always seemed icky to me. Aether uses a linked-list (like world contacts) and Farseer just uses a Queue.

### CODE REVIEW

File: Robust.Shared/Physics/Dynamics/ContactManager.cs [hunk 1]
```diff
@@ -71,14 +72,22 @@ public ContactManager()
...
             ContactCount = 0;
-            ContactPoolList = new ContactHead();
             OnBroadPhaseCollision = AddPair;
         }
 
         public void Initialize()
         {
             IoCManager.InjectDependencies(this);
             _broadPhaseSystem = EntitySystem.Get<SharedBroadPhaseSystem>();
+            InitializePool();
```
Comment (lordpidey) [inline Robust.Shared/Physics/Dynamics/ContactManager.cs:82]: Why was the initialization of ContactPoolList moved from the constructor to the initializer?

Comment (metalgearsloth) [reply Robust.Shared/Physics/Dynamics/ContactManager.cs:82]: Contact has an IEntitySystemManager dependency.

**Review by gradientvera — APPROVED** [review]

**Comment by gradientvera** [comment]
Do you return the contacts to the stack at any point, or?

**Comment by metalgearsloth** [comment]
> 
> 
> Do you return the contacts to the stack at any point, or?

Under Destroy they do

**Comment by gradientvera** [comment]
good for merge then

**Pull request merged**